test/truefi2/TrueFiPool2.test.ts

```diff
@@ -220,13 +268,202 @@ describe('TrueFiPool2', () => {
   })
 
   describe('join-exit', () => {
-    // requires strategy
-    // requires lender
+    beforeEach(async () => {
+      await tusd.approve(pool.address, parseEth(1e7))
+      await pool.join(parseEth(1e7))
+      await tusd.mint(borrower.address, parseEth(1e6))
+      await tusd.connect(borrower).approve(pool.address, parseEth(1e6))
+    })
+
+    it('does not allow to join when joining is paused', async () => {
+      await tusd.approve(pool.address, parseEth(1e6))
+      await pool.changeJoiningPauseStatus(true)
+      await expect(pool.join(parseEth(1e6)))
+        .to.be.revertedWith('TrueFiPool: Joining the pool is paused')
+    })
+
+    it('mints liquidity tokens as 1-to-1 to TUSD for first user', async () => {
+      expect(await pool.balanceOf(owner.address)).to.equal(parseEth(1e7))
+    })
+
+    it('mints liquidity tokens proportionally to stake for next users', async () => {
+      const loan1 = await deployContract(
+        LoanToken2Factory,
+        pool.address,
+        borrower.address,
+        lender.address,
+        AddressZero,
+        parseEth(1e6),
+        DAY * 365,
+        1000,
+      )
+      await rater.mock.getResults.returns(0, 0, parseTRU(15e6))
+      await lender.connect(borrower).fund(loan1.address)
+      await timeTravel(provider, DAY * 182.5)
+      const totalSupply = await pool.totalSupply()
+      const poolValue = await pool.poolValue()
+
+      await pool.connect(borrower).join(parseEth(1e6))
+      expectScaledCloseTo(await pool.balanceOf(borrower.address), totalSupply.mul(parseEth(1e6)).div(poolValue))
+    })
+
+    it('returns a basket of tokens on exit', async () => {
+      const loan1 = await deployContract(
+        LoanToken2Factory,
+        pool.address,
+        borrower.address,
+        lender.address,
+        AddressZero,
+        parseEth(1e6),
+        DAY * 365,
+        1000,
+      )
+      await rater.mock.getResults.returns(0, 0, parseTRU(15e6))
+      await lender.connect(borrower).fund(loan1.address)
+      await timeTravel(provider, DAY * 182.5)
+      const loan2 = await deployContract(
+        LoanToken2Factory,
+        pool.address,
+        borrower.address,
+        lender.address,
+        AddressZero,
+        parseEth(1e6),
+        DAY * 365,
+        2500,
+      )
+      await lender.connect(borrower).fund(loan2.address)
+
+      const liquidValue = await pool.liquidValue()
+      const totalSupply = await pool.totalSupply()
+      const exitAmount = totalSupply.div(2)
+
+      await pool.exit(exitAmount)
+      expect(await tusd.balanceOf(owner.address)).to.equal(exitAmount.mul(liquidValue).div(totalSupply))
+      expectCloseTo(await loan1.balanceOf(owner.address), parseEth(55e4), 10)
+      expectCloseTo(await loan2.balanceOf(owner.address), parseEth(625e3), 10)
```

Comment:
I'm not sure I get where the 55e4 and 625e3 come from. Could you show the calculation?